packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -760,26 +785,35 @@ export class FluidDataStoreRuntime
 	 */
 	public async getGCData(fullGC: boolean = false): Promise<IGarbageCollectionData> {
 		const builder = new GCDataBuilder();
-		// Iterate over each channel context and get their GC data.
-		await Promise.all(
-			Array.from(this.contexts)
-				.filter(([contextId, _]) => {
-					// Get GC data only for attached contexts. Detached contexts are not connected in the GC reference
-					// graph so any references they might have won't be connected as well.
-					return this.isChannelAttached(contextId);
-				})
-				.map(async ([contextId, context]) => {
-					const contextGCData = await context.getGCData(fullGC);
-					// Prefix the child's id to the ids of its GC nodes so they can be identified as belonging to the child.
-					// This also gradually builds the id of each node to be a path from the root.
-					builder.prefixAndAddNodes(contextId, contextGCData.gcNodes);
-				}),
+		await this.visitLocalBoundContextsDuringSummary(
+			async (contextId: string, context: IChannelContext) => {
+				const contextGCData = await context.getGCData(fullGC);
+				// Prefix the child's id to the ids of its GC nodes so they can be identified as belonging to the child.
+				// This also gradually builds the id of each node to be a path from the root.
+				builder.prefixAndAddNodes(contextId, contextGCData.gcNodes);
+			},
 		);
-
 		this.updateGCNodes(builder);
 		return builder.getGCData();
 	}
 
+	/**
+	 * Helper method for preparing to summarize this channel.
+	 * Runs the callback for each bound context to incorporate its data however the caller specifies
+	 */
+	private async visitLocalBoundContextsDuringSummary(
```

Comment:
Rename to `visitContextsDuringSummary`?  The other one filters to bound contexts and asserts they're all local due to the scenario.  I guess we could call this `visitAttachedContextsDuringSummary`, since that's a requirement and we assert otherwise, up to you.

Reply:
Also, consider moving down next to the other visit function